lib/foreman_maintain/object_cache.rb

```diff
@@ -0,0 +1,34 @@
+require 'singleton'
+
+module ForemanMaintain
+  class ObjectCache
+    include Singleton
```

Comment:
Should the ObjectCache be singleton? Different `DependencyGarphs` could have different collections of steps, and having a cache, that would interfere with those might be not desired